executor/batch_point_get.go

```diff
@@ -304,14 +308,27 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 		e.values = append(e.values, val)
 		handles = append(handles, e.handles[i])
 		if e.lock && rc {
+			if !hasKeys {
+				hasKeys = true
+			}
 			existKeys = append(existKeys, key)
 		}
 	}
 	// Lock exists keys only for Read Committed Isolation.
-	if e.lock && rc {
-		err = e.lockKeys(ctx, existKeys)
-		if err != nil {
-			return err
+	if e.lock {
+		if rc {
```

Comment:
We could set the `hasKeys` flag by the condition `if len(existKeys) > 0` before or after the `lockkeys` function call for the RC isolation level.